pkg/webhook/failed_status_checks_add_comment.go

```diff
@@ -54,18 +54,41 @@ func (h *failedStatusCheckAddComment) HandleEvent(w http.ResponseWriter, eventOb
 	commitSHA := event.GetSHA()
 	query := fmt.Sprintf("type:pr state:open repo:%s %s", event.Repo.GetFullName(), commitSHA)
 	searchResult, _, err := gh.Search.Issues(context.Background(), query, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find PR using query %s", query)
+	}
+
+	owner, repo := event.Repo.Owner.GetLogin(), event.Repo.GetName()
+
 	var multiErr error
 	for _, issue := range searchResult.Issues {
 		if issue.PullRequestLinks == nil {
 			continue
 		}
 
-		msg := fmt.Sprintf("Status check _%s_ returned **%s**.", event.GetContext(), state)
+		prNumber, prURL := issue.GetNumber(), issue.GetHTMLURL()
+
+		message := fmt.Sprintf("Status check _%s_ returned **%s**.", event.GetContext(), state)
 		if event.GetTargetURL() != "" {
-			msg += fmt.Sprintf(" See %s for more details.", event.GetTargetURL())
+			message += fmt.Sprintf(" See %s for more details.", event.GetTargetURL())
+		}
+
+		existingComments, _, err := gh.Issues.ListComments(context.Background(), owner, repo, prNumber, &github.IssueListCommentsOptions{
+			Sort:      "updated",
+			Direction: "desc",
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to retrieve existing comments on PR %s", prURL)
+		}
+
+		for _, existingComment := range existingComments {
+			if message == existingComment.GetBody() {
```

Comment:
What is if there is a comment with the same content but happening because of a second (later) check failed ? Is it ok to ommit this comment, too ? If this second comment should still be added maybe one can simply add a html comment `<!-- event: 1245 -->` with the event id to the comment so that only comments coming from the same event are squeezed.